### PR TITLE
Add UUID and Revision number to the API.

### LINKED ISSUE #31: Objects should have a UUID tracking ID and a revision number
Currently, we use a `parent` field, which references ID, to keep track of revisions. Instead, we shall use a UUID tracker, which is constant across revisions of the same object, and a revision number in that object's sequence. The pair (uuid, revision) shall be a unique key for an object.

### PR DESCRIPTION
**Pull request by MorganEPatch**
Fixes #31. Also contains ideas and suggestions for /times/:uuid query string parameters.

### CODE REVIEW

**Comment by tschuy** [comment]
You should use different UUIDs for the project, activity, and time objects.

**Comment by pop** [comment]
refs a comment in #30

File: source/draft_api.rst [hunk 1]
```diff
@@ -54,6 +54,24 @@ a very specific format:
 #) Sets of lowercase letters and numbers can be separated with a single hyphen
 #) Must contain at least one letter
 
+---------
+
+Revisions
+---------
+
+When an object is first created, it is assigned a tracking ID. This a UUID which will
```
Comment (athai) [inline source/draft_api.rst]: "This _is_ a UUID which will..." <-- Missed a word

**Comment by pop** [comment]
When posting to /times/<whatever> we still send to [`/times/<id>`](https://github.com/osuosl/timesync/pull/32/files#diff-582e9e00c01a46b6cf6cfacd61eb6e1eR306) should that be `/times/<uuid>`?
Everything else looks good.

**Comment by pop** [comment]
Looks good to me. +1

**Comment by athai** [comment]
Looks good. +1

**Pull request merged**